src/libraries/System.Net.Quic/src/System.Net.Quic.csproj

```diff
@@ -128,7 +128,7 @@
   </ItemGroup>
 
   <!-- Support for deploying msquic -->
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' and
+  <ItemGroup Condition="'$(_hostOS)' == 'windows' and
```

Comment:
That's an internal property. Don't we have a better one to use? cc @am11

Reply:
Would also be good to add a comment explaining why TargetsWindows can't be used here. I don't know that either yet.

Reply:
Note that `Targets{OSName}` properties in top level Directory.Build.props do not get set for src/libraries, we set them in `eng/targetframeworksuffix.props`. It would be nice to get rid of whole `SkipInferTargetOSName` mechanism and just use the top-level Directory.Build.props, unless it is impossible to consolidate (which I don't think is the case).

Reply:
i can add the comment. but the explaination is 
The non portable build triggers all config build which build this project for net6.0-windows on linux, hence targetsWIndows is true but the runtime is linux so all the ```BinPlaceDir``` are linux directories and thee file gets binplaced with the linux runtime.
i think we want to binplace it in the runtime when we arre building on windows, i added @wfurt to verify that

Reply:
comment would be always good. We should package this only for Windows runtime - regardless where this is build. _if_ `_hostOS` is the right variables I'm fine with that.

Reply:
> That's an internal property.

Correct, properties with `_` indicate "do no use outside of this file" and can break in the future. We have managed to avoid using these internal properties elsewhere. Since this is the first use-case of HostOS outside of `Directory.Build.props`, we can (unconditionally) define `<HostOS>$(_hostOS)</HostOS>` between these two lines: https://github.com/dotnet/runtime/blob/1cd5033bb37d34aecacabb7a4a1543774961a74e/Directory.Build.props#L20-L21
and then in the following lines of Directory.Build.props and here, use `<HostOS>`.

Reply:
Why can't we use `TargetOS` here instead?

Reply:
I think it's because of the non-portable build which @Anipik alluded above: https://github.com/dotnet/runtime/pull/58651#discussion_r705580477
But is this the only place in the entire build infra where it is applicable? If not, what is the alternative way of solving it, maybe we should use that approach to keep it consistent?

Reply:
But the sourcebuild leg sets the OS to Linux in the yml, why would the TargetOS be "Windows" there? https://github.com/dotnet/runtime/blob/3e6d492bdf6fbf2d8af3871379f31dcc6e27716b/eng/pipelines/common/platform-matrix.yml#L191

Reply:
```TargetsWindows = true ``` is true in the innerbuilds when we dispatch innerbuild from net6.0-windows.